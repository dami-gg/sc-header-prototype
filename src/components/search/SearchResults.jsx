import React, {Component} from 'react';

import './Search.scss';

class SearchResults extends Component {
  render() {
    let resultNodes = this.props.results.map((result) => {
      return (
          <li
              className="dc-suggest__item dc-link"
              onClick={() => {
                return this.props.onClickAction && this.props.onClickAction(result);
              }}
              key={result.id}
          >
            {result.name}
          </li>
      );
    });

    return (
        <div>
          <ul className={'dc-list dc-suggest' + (this.props.results.length > 0 ? ' visible' : '')}>
            {resultNodes}
          </ul>
        </div>
    );
  }
}

export default SearchResults;
