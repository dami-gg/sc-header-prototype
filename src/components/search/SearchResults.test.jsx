import React from 'react';
import renderer from 'react-test-renderer';
import SearchResults from './SearchResults';

let mockedSearchResults = [{
  id: 1,
  name: 'Result 1'
}, {
  id: 2,
  name: 'Result 2'
}];

describe('SearchResults component', () => {
  it('should render correctly', () => {
    let tree = renderer.create(<SearchResults searchResults={mockedSearchResults} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});