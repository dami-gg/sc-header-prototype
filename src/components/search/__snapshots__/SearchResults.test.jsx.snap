exports[`SearchResults component should render correctly 1`] = `
<ul
  className="dc-list dc-suggest visible">
  <li
    className="dc-suggest__item dc-link"
    onClick={[Function onClick]}>
    Result 1
  </li>
  <li
    className="dc-suggest__item dc-link"
    onClick={[Function onClick]}>
    Result 2
  </li>
</ul>
`;
